src/components/structures/RoomStatusBar.js

```diff
@@ -285,6 +287,21 @@ module.exports = React.createClass({
         return avatars;
     },
 
+    _shouldShowConnectionError: function() {
+        // no conn bar trumps unread count since you can't get unread messages
+        // without a connection! (technically may already have some but meh)
+        // It also trumps the "some not sent" msg since you can't resend without
+        // a connection!
+        // There's one situation in which we don't show this 'no connection' bar, and that's
+        // if it's a monthly-active-user limit error: those are shown in the top bar.
+        const errorIsMauError = Boolean(
+            this.state.syncStateData &&
+            this.state.syncStateData.error &&
+            this.state.syncStateData.error.errcode === 'M_MAU_LIMIT_EXCEEDED'
+        );
+        return this.state.syncState === "ERROR" && !errorIsMauError;
```

Comment:
Won't this return true in case of a consent error? I presume on connection error there is no actual error object in the syncStateData? Maybe just check for that?

Reply:
The consent error is only for sending messages rather than syncing (we're only worried about the user putting data *into* the system rather than getting it out). The reason for specifying that it's a MAU error is to avoid the situation where we add another failure with a dedicated errcode and an older Riot that doesn't understand that errcode displays no error at all.